addons/calendar/models/calendar.py

```diff
@@ -1232,20 +1239,36 @@ def get_display_time_tz(self, tz=False):
             self = self.with_context(tz=tz)
         return self._get_display_time(self.start, self.stop, self.duration, self.allday)
 
+    def get_origin_meeting_and_schedule_fields(self):
+        """ Given a virtual event record. This method return the original record and its id. It also return schedule fields for the virtual event.
+            Mainly used to avoid code duplication.
+        """
+        real_id = calendar_id2real_id(self.id)
+        meeting_origin = self.browse(real_id)
+
+        data = self.get_schedule_fields()
+        data['start_date' if data['allday'] else 'start_datetime'] = data['start']
+        data['stop_date' if data['allday'] else 'stop_datetime'] = data['stop']
+
+        return real_id, meeting_origin, data
+
+    def _compute_count_division(self, meeting_origin, reccurent_dates):
+        """ Used in get_split_recurring_event. In case of a recurring calendar with count as end_type, we should
+            split the count in two part. Number of events before the selected date and remaining events after it.
+        """
+        start = pytz.UTC.localize(fields.Datetime.from_string(self.start)).astimezone(pytz.UTC)
+        meeting_number = reccurent_dates.index(start)
```

Comment:
No because if the previous call find an error it returns the real_id so we know that this function will not fail since it passed before